tools/internal/autoupdate/registry.go

```diff
@@ -118,118 +93,73 @@ func (r *Registry) Validate() error {
 	return nil
 }
 
-func (r *Registry) getImageTags(githubToken, nextURL string, page int) ([]string, error) {
-	registry, namespace, repository := r.getRegistryInformationFromImage()
-
-	req, err := buildRequest(registry, namespace, repository, fmt.Sprintf("%d", page), nextURL, githubToken)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create request: %w", err)
-	}
-
-	resp, err := doRequestWithRetries(req)
+func (r *Registry) getImageTags() ([]string, error) {
+	registry, err := r.getRegistryInformationFromImage()
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to get registry information from image: %w", err)
 	}
-	defer resp.Body.Close()
-
-	body, err := io.ReadAll(resp.Body)
-	if err != nil {
-		return nil, fmt.Errorf("failed to read response body: %w", err)
-	}
-
-	return r.processRegistryResponse(body, resp.Header.Get("Link"), githubToken, registry, namespace, repository, page)
+	return registry.getImageTags()
 }
 
-func buildRequest(registry, namespace, repository, page, nextUrl, githubToken string) (*http.Request, error) {
-	params := url.Values{}
-	var registryUrl, token string
+func (r *Registry) getRegistryInformationFromImage() (ImageRegistry, error) {
+	image := r.Images[0].SourceImage
+	splittedImage := strings.Split(image, "/")
+	var registry, namespace, repository string
+	// Case 1: Handle default Docker Hub images like "flannel/flannel"
+	if len(splittedImage) == 2 && !strings.Contains(splittedImage[0], ".") {
+		registry = "dockerhub"
+		namespace = splittedImage[0]
+		repository = splittedImage[1]
+	} else if len(splittedImage) == 2 && strings.Contains(splittedImage[0], ".") {
+		// Case 2: Handle images with a registry but no namespace like "k8s.gcr.io/pause"
+		registry = splittedImage[0]
+		namespace = ""
+		repository = splittedImage[1]
+	} else if len(splittedImage) > 3 && strings.Contains(splittedImage[0], ".") {
```

Comment:
I think that this case can be combined with the "standard 3-part" case. If the image has 3 parts, then `strings.Join(splittedImage[2:], "/")` should simply return `splittedImage[2]`.

Reply:
Good catch!